lib/application.js

```diff
@@ -567,3 +571,65 @@ app.listen = function(cb) {
 
   return server;
 };
+
+/* START CONTROLLER CODE*/
+
+// this is similar to app.model
+// at the moment I haven't really used 'options'
+app.controller = function(ctor, options) {
+  var self = this;
+
+  // the SharedController is a wrapper, see below
+  var controller = new SharedController(ctor);
+  this.remotes().addClass(controller.sharedClass);
+  app.controllers().push(controller);
+
+  return controller;
+
+  // SharedController class contains strong-remoting objects to set up the metadata
+  // and register the endpoint, etc
+  function SharedController(ctor, options) {
+    options = options || {};
+    this.name = ctor.name;
+    this.ctor = ctor;
+    // TODO - right now there's a hack on line 289 to skip this
+    // util.inherits(ctor, EventEmitter);
+
+    this.sharedClass = new RemoteObjects.SharedClass(this.name, ctor, options);
```

Comment:
As per [@bajtos' comment](https://github.com/strongloop/loopback-sandbox/pull/7/files/6c4d633f9b5b55710980069a4efaee2ebe06c2c1#r63170059):
> why would you like to set the controller name in the first place, I mean what do you use the name for?

By default, the path that the SharedClass is registered to the path `'/' + this.name`. The user must set `ctor.http.path` to explicitly define the path at which to register the controller.